roles/openshift_logging_elasticsearch/tasks/main.yaml

```diff
@@ -450,6 +459,57 @@
     es_deploy_name: "{{ openshift_logging_elasticsearch_deployment_name }}"
   when: openshift_logging_elasticsearch_deployment_name != ""
 
+- set_fact:
+    _es_container_security_context: {'privileged': true}
+  when:
+  - openshift_logging_elasticsearch_storage_type == "hostmount"
+
+- set_fact:
+    _es_container_security_context: "{{ _es_containers.elasticsearch.securityContext }}"
+  when:
+  - _es_containers is defined
+  - '"elasticsearch" in _es_containers'
+  - '"securityContext" in _es_containers.elasticsearch'
+
+- set_fact:
+    _es_node_selector: "{{ openshift_logging_elasticsearch_nodeselector | default({}) }}"
+  when:
+  - openshift_logging_elasticsearch_storage_type != "hostmount"
```

Comment:
Why would this be required? `_es_node_selector` is being set in both cases

Reply:
`_es_node_selector` is used in the following task "Set ES dc templates". Depending on `openshift_logging_elasticsearch_storage_type == "hostmount"` it is set to `openshift_logging_elasticsearch_nodeselector` or `openshift_logging_elasticsearch_hostmount_nodeselector`.

Reply:
`_es_node_selector` is being set to `openshift_logging_elasticsearch_nodeselector` and at line 484. Lets just use `{{ openshift_logging_elasticsearch_nodeselector | default({}) }}` at line 527

Reply:
@vrutkovs, the variables in this two lines are similiar `openshift_logging_elasticsearch_nodeselector ` and `openshift_logging_elasticsearch_**hostmount**_nodeselector`. Are you considering they are the same when analyzing this code snippet? Or am I not understand you?

The logic is:
```
if openshift_logging_elasticsearch_storage_type != "hostmount"
  _es_node_selector = openshift_logging_elasticsearch_nodeselector 
else
  _es_node_selector = openshift_logging_elasticsearch_hostmount_nodeselector

Then create template with 
  es_node_selector = _es_node_selector 
```

Reply:
Why would a dedicated var `openshift_logging_elasticsearch_hostmount_nodeselector` for hostmount be required? If hostmount is being used the admin would just update `openshift_logging_elasticsearch_nodeselector`

Reply:
@vrutkovs, maybe the variable name is not good. It might suggests that admin could use it to define schedulable nodes for ES. But it's meant to be use only internally, only by the installer.

Admin defines the set of all schedulable nodes in `openshift_logging_elasticsearch_nodeselector` and when installer allocates one exclusive node that the DC while be pinned, it uses this variable `openshift_logging_elasticsearch_nodeselector` to hold this info.

But I agreed this might be confusing and I reimplemented this code slightly different, with the same logic. Could you check if it is better now?